std-bits/base/src/main/java/org/enso/base/Text_Utils.java

```diff
@@ -153,14 +238,67 @@ public static String from_codepoints(int[] codepoints) {
     return new String(codepoints, 0, codepoints.length);
   }
 
+  private static CharBuffer resize(CharBuffer old) {
```

Comment:
DRY?

Reply:
felt the generic solution wasn't very clean:
```java
  private static <T extends Buffer> T resize(T old, IntFunction<T> allocate, BiConsumer<T, T> put) {
    int n = old.capacity();
    int new_n = 2*n + 1;
    T o = allocate.apply(new_n);
    old.flip();
    put.accept(o, old);
    return o;
  }
```

as Buffer doesn't have `put` or `allocate`. But happy to switch to generic if you think better.

Reply:
I would actually argue for having the two specializations.
Maybe it is premature optimization, but feels like having `ByteBuffer` or `CharBuffer` directly can allow the JVM to statically resolve the methods to be used whereas having the closures it may not realise this is possible, at least that quickly. And it may be important because encoding is quite a primitive operation, which can work with quite big amounts of data and so performance is valuable here.

But if @hubertp you think it shouldn't matter much (I don't have enough JVM experience to judge if the overhead of closures here is noticeable or not), then I agree that a generic solution is more elegant.

Reply:
don't think it will have any performance overhead as just method references being passed - does make it slightly cleaner to only have the method once. 

The `Buffer` class is ancient so not thought about for this stuff!